Source/WebCore/platform/PublicSuffixStore.cpp

```diff
@@ -96,4 +96,33 @@ String PublicSuffixStore::topPrivatelyControlledDomain(StringView host) const
     return result;
 }
 
+String PublicSuffixStore::topPrivatelyControlledDomainWithoutPublicSuffix(StringView host) const
+{
+    auto topPrivatelyControlledDomain = this->topPrivatelyControlledDomain(host);
+    if (topPrivatelyControlledDomain.isEmpty())
+        return { };
+
+    return domainWithoutPublicSuffix(topPrivatelyControlledDomain);
+}
+
+String PublicSuffixStore::domainWithoutPublicSuffix(StringView domain) const
+{
+    if (URL::hostIsIPAddress(domain))
+        return domain.toString();
+
+    size_t separatorPosition;
+    for (unsigned labelStart = 0; (separatorPosition = domain.find('.', labelStart)) != notFound; labelStart = separatorPosition + 1) {
+        auto candidate = domain.substring(separatorPosition + 1);
+        if (isPublicSuffix(candidate)) {
+            if (separatorPosition > 0 && domain.length() > separatorPosition) {
+                auto domainWithoutSuffix = domain.substring(0, separatorPosition);
+                auto foo = domainWithoutSuffix.toString();
```

Comment:
Seems like this can be deleted.

Reply:
Oh yes -- stupid debugging code I forgot about. Will fix.